packages/protocol/contracts/governance/BondedDeposits.sol

```diff
@@ -1,5 +1,6 @@
 pragma solidity ^0.5.8;
 
+import "fixidity/contracts/FixidityLib.sol";
```

Comment:
Just looked at this library, it looks *very* simple and unnecessarily complex at the same time.

Since we don't care about negative numbers, one option would be to fix the denominator as 2 ^ 64, and make the max numerator 2 ^ 128. This way we could still do multiplications within 256 bits, and would be able to represent numbers as high as 2 ^ 64.

What does fixidity give us that such a scheme would not?

Reply:
1. It's already written.
2. They also have exponentiation/roots/logarithms that we could potentially use elsewhere if we like their implementation.

With that being said, I'm open to using other solutions. Looks like the author of Fixidity himself [recommends](https://medium.com/hackernoon/10x-better-fixed-point-math-in-solidity-32441fd25d43) a [different library](https://github.com/abdk-consulting/abdk-libraries-solidity), though this one has a 64-64 split on `int128`s, which feels unnecessarily limiting.

Could also roll our own, potentially improving upon/simplifying Fixidity.

Reply:
Hmm, if the author recommends against using their library we may want to listen.

I'm not sure we can use ADBK for license reasons (all rights reserved).

What do you see as the pros and cons of rolling our own?

Reply:
Cons: time, potential of bugs. So really the same cons as any change to a codebase :P. But here potential of bugs is critical since it touches everything and we've already seen that it's easy to miss problems with a simple-looking library.

Pros: we can probably get it cleaner, more efficient than Fixidity. Hopefully could at least partially address your concerns about increased gas costs - this is most likely caused by the fact this is a linked library which we could potentially change to inlined and save gas on calls.